docs/API.md

```diff
@@ -0,0 +1,176 @@
+* parameters - a dictionary of the parameters configuring the action.
+* tracking info - (user, time, etc)
+* action_lifecycle value:
+  * Pending
+  * Validation Failed
+  * In DAG
+  * Complete
+  * Failed
+* DAG_status - representing the status that airflow provides for an executing DAG
+* validations - a list of validaitons that have been done, including any status information for those validations. During the lifecycle of the DAG, this list of validations may continue to grow.
+* steps - the list of steps for this action, including the status for that step.
+
+#### GET
+Returns the list of actions in the system that have been posted, and are accessible to the current user.
+
+##### Responses
+* 200 OK  
+If the actions can be retrieved.
+
+#### POST
+Creates an action in the system. This will cause some action to start.
+The input body to this post will represent an action object that has at least these fields:
+* name  
+The name of the action to invoke. This is likely to map to the actual DAG Names, but may be mapped for "nicer" values.
+* parameters  
+A dictionary of parameters to use for the trigger invocation. The supported parameters will vary for the action invoked.
+
+The POST will synchrounously create the action (a shell object that represents a DAG invocation), perform any checks to validate the preconditions to run the DAG, and trigger the invocation of the DAG.
+The DAG will run Asynchrounously in airflow.
+##### Responses
+* 201 Created  
+If the action is created successfully, and all preconditions to run the DAG are successful. The response body is the action entity created.  
+* 400 Bad Request  
+If the action name doesn't exist, or the payload is otherwise malformed.  
+* 409 Conflict  
+For any failed validations. The response body is the action entity created, with the failed validations. The DAG will not begin execution in this case.  
+
+
+### /v1.0/actions/{action_id} 
+Each action will be assigned an unique id that can be used to get details for the action, including the execution status.
+
+#### GET
+Returns the action entity for the specified id.
+##### Responses
+* 200 OK 
+
+
+### /v1.0/actions/{action_id}/validationDetails/{validation_id} 
+Allows for drilldown to validation detailed info.
+
+#### GET
+Returns the validation detail by Id for the supplied action Id.
+##### Responses
+* 200 OK 
+
+
+### /v1.0/actions/{action_id}/steps/{step_id}
+Allow for drilldown to step information. The step information includes details of the steps excution, successful or not, and enough to facilitate troubleshooting in as easy a fashion as possible. 
+
+#### GET
+Returns the details for a step by id for the given action by Id.
+##### Responses
+* 200 OK 
+
+
+### /v1.0/actions/{action_id}/{control_verb}
+Allows for issuing DAG controls against an action.
+
+#### POST 
+Trigger a control action against an activity.- this includes: pause, unpause, abandon
+##### Responses
```

Comment:
This needs more discussion - the working concept with abandon is that we will need to have subdags that encapsulate a few steps into a unit of work that starts by having a check against some external source to determine if the flow should instead go to a graceful exit path. What that graceful exit path means is not yet determined, but could be a complex backout or cleanup. For near term, I expect that we will only have pause, unpause, and any abandon path will be a future enhancement when we've had enough time to address the potential fallout.  I'm removing abandon for now.